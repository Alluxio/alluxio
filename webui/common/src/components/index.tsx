/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

export * from './HOCs/withLoadingMessage/withLoadingMessage';
export * from './HOCs/withErrors/withErrors';
export * from './HOCs/withTextAreaResize/withTextAreaResize';
export * from './HOCs/withFetchDataFromPath/withFetchDataFromPath';
export * from './HOCs/withFetchData/withFetchData';
export * from './HOCs/withFluidContainer/withFluidContainer';
export * from './Footer/Footer';
export * from './Header/Header';
export * from './LoadingMessage/LoadingMessage';
export * from './Logs/Logs';
export * from './FileView/FileView';
export * from './Paginator/Paginator';
export * from './LineGraph/LineGraph';
