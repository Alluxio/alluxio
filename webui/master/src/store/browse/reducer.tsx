/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {Reducer} from 'redux';

import {BrowseActionTypes, IBrowseState} from './types';

export const initialBrowseState: IBrowseState = {
  data: {
    'accessControlException': '',
    'blockSizeBytes': '',
    'currentDirectory': {
      'absolutePath': '',
      'blockSizeBytes': '',
      'blocksOnTier': {},
      'creationTime': '',
      'fileLocations': [],
      'group': '',
      'id': 0,
      'inAlluxio': false,
      'inAlluxioPercentage': 0,
      'isDirectory': true,
      'mode': '',
      'modificationTime': '',
      'name': '',
      'owner': '',
      'persistenceState': '',
      'pinned': false,
      'size': ''
    },
    'currentPath': '',
    'debug': false,
    'fatalError': '',
    'fileBlocks': [],
    'fileData': '',
    'fileDoesNotExistException': '',
    'fileInfos': [],
    'highestTierAlias': '',
    'invalidPathError': '',
    'invalidPathException': '',
    'masterNodeAddress': '',
    'ntotalFile': 0,
    'pathInfos': [],
    'showPermissions': true,
    'viewingOffset': 0
  },
  errors: undefined,
  loading: false,
};

export const browseReducer: Reducer<IBrowseState> = (state = initialBrowseState, action) => {
  switch (action.type) {
    case BrowseActionTypes.FETCH_REQUEST:
      return {...state, loading: true};
    case BrowseActionTypes.FETCH_SUCCESS:
      return {...state, loading: false, data: action.payload.data, response: action.payload, errors: undefined};
    case BrowseActionTypes.FETCH_ERROR:
      return {...state, loading: false, errors: action.payload};
    default:
      return state;
  }
};
