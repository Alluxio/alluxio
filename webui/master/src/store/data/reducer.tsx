/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {Reducer} from 'redux';

import {DataActionTypes, IDataState} from './types';

export const initialDataState: IDataState = {
  data: {
    'fatalError': '',
    'fileInfos': [],
    'inAlluxioFileNum': 0,
    'masterNodeAddress': '',
    'permissionError': '',
    'showPermissions': true
  },
  errors: undefined,
  loading: false
};

export const dataReducer: Reducer<IDataState> = (state = initialDataState, action) => {
  switch (action.type) {
    case DataActionTypes.FETCH_REQUEST:
      return {...state, loading: true};
    case DataActionTypes.FETCH_SUCCESS:
      return {...state, loading: false, data: action.payload.data, response: action.payload, errors: undefined};
    case DataActionTypes.FETCH_ERROR:
      return {...state, loading: false, errors: action.payload};
    default:
      return state;
  }
};
