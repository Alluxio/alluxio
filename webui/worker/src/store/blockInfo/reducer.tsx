/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {Reducer} from 'redux';

import {BlockInfoActionTypes, IBlockInfoState} from './types';

export const initialBlockInfoState: IBlockInfoState = {
  data: {
    'blockSizeBytes': '',
    'fatalError': '',
    'fileBlocksOnTier': [],
    'fileInfos': [],
    'invalidPathError': '',
    'ntotalFile': 0,
    'orderedTierAliases': [],
    'path': ''
  },
  errors: undefined,
  loading: false
};

export const blockInfoReducer: Reducer<IBlockInfoState> = (state = initialBlockInfoState, action) => {
  switch (action.type) {
    case BlockInfoActionTypes.FETCH_REQUEST:
      return {...state, loading: true};
    case BlockInfoActionTypes.FETCH_SUCCESS:
      return {...state, loading: false, data: action.payload.data, response: action.payload, errors: undefined};
    case BlockInfoActionTypes.FETCH_ERROR:
      return {...state, loading: false, errors: action.payload};
    default:
      return state;
  }
};
