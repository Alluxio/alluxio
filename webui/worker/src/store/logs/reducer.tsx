/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {Reducer} from 'redux';

import {ILogsState, LogsActionTypes} from './types';

export const initialLogsState: ILogsState = {
  data: {
    'currentPath': '',
    'debug': false,
    'fatalError': '',
    'fileData': '',
    'fileInfos': [],
    'invalidPathError': '',
    'ntotalFile': 0,
    'viewingOffset': 0
  },
  errors: undefined,
  loading: false
};

export const logsReducer: Reducer<ILogsState> = (state = initialLogsState, action) => {
  switch (action.type) {
    case LogsActionTypes.FETCH_REQUEST:
      return {...state, loading: true};
    case LogsActionTypes.FETCH_SUCCESS:
      return {...state, loading: false, data: action.payload.data, response: action.payload, errors: undefined};
    case LogsActionTypes.FETCH_ERROR:
      return {...state, loading: false, errors: action.payload};
    default:
      return state;
  }
};
