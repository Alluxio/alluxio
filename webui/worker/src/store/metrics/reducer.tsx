/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {Reducer} from 'redux';

import {IMetricsState, MetricsActionTypes} from './types';

export const initialMetricsState: IMetricsState = {
  data: {
    'operationMetrics': {},
    'rpcInvocationMetrics': {},
    'timeSeriesMetrics': {},
    'workerCapacityFreePercentage': 0,
    'workerCapacityUsedPercentage': 0
  },
  errors: undefined,
  loading: false
};

export const metricsReducer: Reducer<IMetricsState> = (state = initialMetricsState, action) => {
  switch (action.type) {
    case MetricsActionTypes.FETCH_REQUEST:
      return {...state, loading: true};
    case MetricsActionTypes.FETCH_SUCCESS:
      return {...state, loading: false, data: action.payload.data, response: action.payload, errors: undefined};
    case MetricsActionTypes.FETCH_ERROR:
      return {...state, loading: false, errors: action.payload};
    default:
      return state;
  }
};
