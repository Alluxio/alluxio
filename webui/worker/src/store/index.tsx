/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {connectRouter, RouterState} from 'connected-react-router';
import {History} from 'history';
import {combineReducers} from 'redux';
import {all, fork} from 'redux-saga/effects';

import {initialRefreshState, refreshReducer} from '@alluxio/common-ui/src/store/refresh/reducer';
import {IRefreshState} from '@alluxio/common-ui/src/store/refresh/types';
import {initialLogsState, logsReducer} from '@alluxio/common-ui/src/store/logs/reducer';
import {logsSaga} from '@alluxio/common-ui/src/store/logs/sagas';
import {ILogsState} from '@alluxio/common-ui/src/store/logs/types';
import {blockInfoReducer, initialBlockInfoState} from './blockInfo/reducer';
import {blockInfoSaga} from './blockInfo/sagas';
import {IBlockInfoState} from './blockInfo/types';
import {initialMetricsState, metricsReducer} from './metrics/reducer';
import {metricsSaga} from './metrics/sagas';
import {IMetricsState} from './metrics/types';
import {initialOverviewState, overviewReducer} from './overview/reducer';
import {overviewSaga} from './overview/sagas';
import {IOverviewState} from './overview/types';
import {IInitState} from './init/types';
import {initialInitState, initReducer} from './init/reducer';
import {initSaga} from './init/sagas';

export interface IApplicationState {
  blockInfo: IBlockInfoState;
  init: IInitState;
  logs: ILogsState;
  metrics: IMetricsState;
  overview: IOverviewState;
  refresh: IRefreshState;
  router?: RouterState;
}

export const rootReducer = (history: History) => combineReducers<IApplicationState>({
  blockInfo: blockInfoReducer,
  init: initReducer,
  logs: logsReducer,
  metrics: metricsReducer,
  overview: overviewReducer,
  refresh: refreshReducer,
  router: connectRouter(history)
});

export const rootSaga = function* () {
  yield all([
    fork(blockInfoSaga),
    fork(initSaga),
    fork(logsSaga),
    fork(metricsSaga),
    fork(overviewSaga)
  ]);
};

export const initialState: IApplicationState = {
  blockInfo: initialBlockInfoState,
  init: initialInitState,
  logs: initialLogsState,
  metrics: initialMetricsState,
  overview: initialOverviewState,
  refresh: initialRefreshState
};
