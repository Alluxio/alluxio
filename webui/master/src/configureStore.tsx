/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {routerMiddleware} from 'connected-react-router';
import {History} from 'history';
import {applyMiddleware, createStore, Store} from 'redux';
import {composeWithDevTools} from 'redux-devtools-extension';
import createSagaMiddleware from 'redux-saga';

import {IApplicationState, rootReducer, rootSaga} from './store';

export default function configureStore(history: History, initialState: IApplicationState): Store<IApplicationState> {
  const composeEnhancers = composeWithDevTools({});
  const sagaMiddleware = createSagaMiddleware();

  const store = createStore(
    rootReducer(history),
    initialState,
    composeEnhancers(applyMiddleware(routerMiddleware(history), sagaMiddleware))
  );

  sagaMiddleware.run(rootSaga);
  return store;
}
