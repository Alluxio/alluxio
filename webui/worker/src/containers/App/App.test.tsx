/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {configure, mount, ReactWrapper, shallow, ShallowWrapper} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {createBrowserHistory, History, LocationState} from 'history';
import React from 'react';
import {Provider} from 'react-redux';
import {Store} from 'redux';
import sinon, {SinonSpy} from 'sinon';

import {initialRefreshState} from '@alluxio/common-ui/src/store/refresh/reducer';
import configureStore from '../../configureStore'
import {initialState, IApplicationState} from '../../store';
import {initialInitState} from '../../store/init/reducer';
import ConnectedApp, {AllProps, App} from './App';
import {Footer, Header} from "@alluxio/common-ui/src/components";
import {routePaths} from "../../constants";
import {createAlertErrors} from "@alluxio/common-ui/src/utilities";

configure({adapter: new Adapter()});

describe('App', () => {
  let history: History<LocationState>;
  let store: Store<IApplicationState>;
  let props: AllProps;

  beforeAll(() => {
    history = createBrowserHistory({keyLength: 0});
    history.push(routePaths.root);
    store = configureStore(history, initialState);
    props = {
      history: history,
      init: initialInitState.data,
      triggerRefresh: sinon.spy(() => {}),
      class: '',
      refresh: initialState.refresh.data,
      errors: createAlertErrors(false),
      loading: false
    };
  });

  describe('Shallow component', () => {
    let shallowWrapper: ShallowWrapper;

    beforeAll(() => {
      shallowWrapper = shallow(<App {...props}/>);
    });

    it('Renders without crashing', () => {
      expect(shallowWrapper.length).toEqual(1);
    });

    it('Should render a Header', () => {
      expect(shallowWrapper.find(Header)).toHaveLength(1);
    });

    it('Should render a Footer', () => {
      expect(shallowWrapper.find(Footer)).toHaveLength(1);
    });

    Object.values(routePaths).forEach(path => {
      it(`Should render Route for ${path}`, () => {
        expect(shallowWrapper
            .findWhere(n => n.name() === 'Route' && n.prop('path') === path))
            .toHaveLength(1);
      });
    });

    it('Should render a Route for redirects', () => {
      expect(shallowWrapper
          .findWhere(n => n.name() === 'Route' && n.prop('path') === undefined))
          .toHaveLength(1);
    });

    it('Matches snapshot', () => {
      expect(shallowWrapper).toMatchSnapshot();
    });
  });
});
