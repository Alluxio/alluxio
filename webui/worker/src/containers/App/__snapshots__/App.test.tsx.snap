// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Shallow component Matches snapshot 1`] = `
<Connect(ConnectedRouterWithContext)
  history={
    Object {
      "action": "PUSH",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 2,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <div
    className="App h-100"
  >
    <div
      className="w-100 sticky-top header-wrapper"
    >
      <Header
        autoRefreshCallback={[Function]}
        callbackParameters={
          Object {
            "masterHostname": "",
            "masterPort": 19999,
          }
        }
        data={
          Array [
            Object {
              "innerText": "Overview",
              "url": "/overview",
            },
            Object {
              "innerText": "BlockInfo",
              "url": "/blockInfo",
            },
            Object {
              "innerText": "Logs",
              "url": "/logs",
            },
            Object {
              "innerText": "Metrics",
              "url": "/metrics",
            },
            Object {
              "innerText": "Return to Master",
              "url": [Function],
            },
          ]
        }
        history={
          Object {
            "action": "PUSH",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 2,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      />
    </div>
    <div
      className="w-100 pt-5 mt-3 pb-4 mb-2"
    >
      <Switch>
        <Route
          exact={true}
          path="/"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/overview"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/blockInfo"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/logs"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/metrics"
          render={[Function]}
        />
        <Route
          render={[Function]}
        />
      </Switch>
    </div>
    <div
      className="w-100 footer-wrapper"
    >
      <Footer
        callbackParameters={
          Object {
            "masterHostname": "",
            "masterPort": 19999,
          }
        }
        data={
          Array [
            Object {
              "attributes": Object {
                "target": "_blank",
              },
              "innerText": "Project Website",
              "url": "https://alluxio.io/",
            },
            Object {
              "attributes": Object {
                "target": "_blank",
              },
              "innerText": "User Mailing List",
              "url": "https://groups.google.com/forum/#!forum/alluxio-users",
            },
            Object {
              "innerText": "Workers",
              "url": [Function],
            },
            Object {
              "attributes": Object {
                "target": "_blank",
              },
              "innerText": "Slack Channel",
              "url": "https://www.alluxio.io/slack",
            },
          ]
        }
      />
    </div>
  </div>
</Connect(ConnectedRouterWithContext)>
`;
